repaygithub/ui-tools
# CACTUS-7 :: End to End Tests

Ticket: https://repayonline.atlassian.net/browse/CACTUS-7

Should we also export the `runEteTests` JS function? It'd be easy enough if we wanted to.

### Testing
1. Check out this branch, navigate to the `testing-tools` module, and run `yarn install`.
2. Publish `testing-tools` with `yalc publish`.
3. In cactus, navigate to the `mock-ebpp` example app.
4. Add `testing-tools` with `yalc add --dev @repay/testing-tools` (you may need to run `yarn add --dev @repay/testing-tools` first, I'm not sure).
5. Run `yarn install`.
6. In `package.json`, change the `test` script to `yarn build && testing-tools run-ete-tests --src tests/**/*.test.ts`
7. In `package.json`, change the `test:local` script to `yarn test:types && yarn test -b browserstack:IE@11:Windows 10 -b browserstack:Chrome:Windows 10`
8. Remove the `src`, `assertionTimeout`, `selectorTimeout`, and `clientScripts` fields from `.testcaferc.json` so that the only field left is the `tsConfigPath`.
9. Run `yarn test:local` and make sure all of the tests run as you'd expect them to.
